lib/bap_image/bap_llvm_binary_stubs.c

```diff
@@ -0,0 +1,157 @@
+#include <caml/memory.h>
+#include <caml/fail.h>
+#include <caml/alloc.h>
+#include <caml/custom.h>
+#include <caml/bigarray.h>
+#include <caml/compatibility.h>
+#include <inttypes.h>
+#include <stdbool.h>
+
+#include "bap_llvm_binary_stubs.h"
+#include "bap_llvm_binary.h"
+
+Noreturn void llvm_binary_fail(const char* message) {
+    caml_failwith(message);
+}
+
+static const struct image* image_from_value(value v) {
+    return *(const struct image**) Data_custom_val(v);
+}
+
+static void custom_finalize_image(value v) {
+    CAMLparam1(v);
+    image_destroy(image_from_value(v));
+    CAMLreturn0;
+}
+
+static value image_to_value(const struct image* img) {
+    CAMLparam0();
+    CAMLlocal1(v);
+    static struct custom_operations binary_ops = {
+        "llvm.caml.llvm_image",
```

Comment:
it is better to use "bap.llvm_loader" instead of prefixing with "llvm", that may introduce clashes (very unlikely though) with real llvm bindings